release.sh

```diff
@@ -5,6 +5,38 @@ cd $DIR
 
 set -e
 
+git remote update
+
+upstream=${1:-'@{u}'}
+remote=$(git rev-parse "$upstream")
+local=$(git rev-parse @)
+
+# check if local commit syncs with remote 
+if [ "$remote" != "$local" ]
+    then
+        echo "Error: local commit does not match remote. Exiting release script." 
+        exit 1
+fi 
+
+remoteTags=$(mktemp)
+git ls-remote --tags origin | grep -v '{}' | sort | tr -d [[:blank:]] >> $remoteTags
+
+localTags=$(mktemp)
+git show-ref --tags | grep -v '{}' | grep -v 'origin' | sort | tr -d [[:blank:]] >> $localTags
```

Comment:
I think you want `grep -v "$upstream"`, right?

Could you also comment on what are doing here?

Reply:
Yeah, this definitely needs a comment.

Reply:
Would it be better to just grep for tags of the form `v[0-9]+\.[0-9]+\.[0-9]+` here, and exclude all the others?

Reply:
@wolffcm, perhaps. Let me try it and see what happens